linera-base/src/data_types.rs

```diff
@@ -842,12 +846,34 @@ impl HashedBlob {
     }
 }
 
+#[derive(Serialize, Deserialize)]
+#[serde(rename = "HashedBlob")]
+struct SerializableBlob {
+    pub blob_type: BlobType,
+    pub blob: Blob,
+}
+
 impl Serialize for HashedBlob {
     fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
     where
-        S: serde::Serializer,
+        S: Serializer,
     {
-        self.blob.serialize(serializer)
+        if serializer.is_human_readable() {
+            let bytes = bcs::to_bytes(&SerializableBlob {
+                blob_type: self.id.blob_type,
+                blob: self.blob.clone(),
+            })
+            .map_err(serde::ser::Error::custom)?;
+            serializer.serialize_str(&hex::encode(bytes))
```

Comment:
Should the blob type be the human-readable string then, and only the payload hex? Something like `Data:0123456789abcdef` and `Bytecode:0123456789abcdef`.

Reply:
I think that's okay. It makes sense to me that a data blob ID looks like `Data:<short hex>` and the blob is `Data:<long hex>`.
(Not sure how useful human-readable string representation of big data blobs will be in the first place. Maybe we should even use base64 instead of hex to make them a bit shorter? But we can always change that later.)